### PR TITLE
Refactor the login tests

### PR DESCRIPTION
**Pull request by josericardo**
I felt the tests intents were lost among the mock machinery.

### CODE REVIEW

**Comment by josericardo** [comment]
@eliasdorneles how can I improve it?

File: tests/test_login.py [hunk 1]
```diff
@@ -19,63 +17,63 @@ def test_login_writes_input_key_to_netrc_file(self):
...
+            for path, content in (files or {}).iteritems():
+                with open(path, 'w') as f:
+                    f.write(content)
 
-    def test_login_attempt_after_login_doesnt_lead_to_an_error(self):
-        with self.runner.isolated_filesystem() as fs:
+        def invoke():
+            return self.runner.invoke(login.cli, input=user_input, env=env)
+
+        def run():
```
Comment (josericardo) [inline tests/test_login.py]: I know, this is smart method. Is it worth it?

Comment (eliasdorneles) [inline tests/test_login.py]: I'm not sure how I feel about this.
I kinda like the reading of the other tests, but this feels a bit like sweeping the dirty under the rug. :)

Comment (josericardo) [inline tests/test_login.py]: The problem is that the login command is just code that reads/writes files, input and makes network requests basically, there isn't much logic. 

We could try a different design for the login, more OO, for example. But I'm not sure it's worth the trouble right now as the feature is already pretty stable.

At least in the refactored code the test purpose is more obvious, IMO.

**Comment by pablohoffman** [comment]
how do make a final call on this @josericardo @eliasdorneles?. do we need a third reviewer?

**Comment by eliasdorneles** [comment]
No need, we've discussed it here and we're merging. :+1:

**Pull request merged**